evm/src/fuzz/invariant/executor.rs

```diff
@@ -551,17 +554,28 @@ fn collect_data(
 
 /// Verifies that the invariant run execution can continue.
 /// Returns the mapping of (Invariant Function Name -> Call Result) if invariants were asserted.
+#[allow(clippy::too_many_arguments)]
 fn can_continue(
     invariant_contract: &InvariantContract,
     call_result: RawCallResult,
     executor: &Executor,
     calldata: &[BasicTxDetails],
     failures: &mut InvariantFailures,
+    targeted_contracts: &FuzzRunIdentifiedContracts,
+    state_changeset: StateChangeset,
     fail_on_revert: bool,
     shrink_sequence: bool,
 ) -> (bool, Option<BTreeMap<String, RawCallResult>>) {
     let mut call_results = None;
-    if !call_result.reverted {
+
+    // Detect handler assertion failures first.
+    let handlers_failed = targeted_contracts
+        .lock()
+        .iter()
+        .any(|contract| !executor.is_success(*contract.0, false, state_changeset.clone(), false));
```

Comment:
even if the contract does not inherit from DSTest this seems to work well (it should be a noop essentially).